pennylane/math/single_dispatch.py

```diff
@@ -167,6 +177,20 @@ def _take_autograd(tensor, indices, axis=None):
 )
 
 
+def _sqrt_matrix_autograd(matrix):
+    """Sqrt matrix for Autograd"""
+    evs, vecs = _i("autograd").numpy.linalg.eigh(matrix)
+    evs = _i("autograd").numpy.where(evs > 0.0, evs, 0.0)
+    return (
+        vecs
+        @ _i("autograd").numpy.diag(_i("autograd").numpy.sqrt(evs))
+        @ _i("autograd").numpy.conj(_i("autograd").numpy.transpose(vecs))
+    )

```

Comment:
I'm against having these different `sqrt_matrix` functions for all the interfaces here, especially since i) they're more complicated than most other `qml.math` functions and ii) they're mostly doing the same thing.

I think it would be better to implement the helper functions such as `eigh` and `diag` here (if they're not already implemented, which you should check), then move the `sqrt_matrix` function to a higher-level file that just calls the `qml.math` functions.

Reply:
Moved from `single_dispatch.py` to `quantum.py`.